src/java/htsjdk/samtools/cram/build/CramCompression.java

```diff
@@ -0,0 +1,28 @@
+package htsjdk.samtools.cram.build;
+
+import htsjdk.samtools.cram.structure.CompressionHeader;
+import htsjdk.samtools.cram.structure.CramCompressionRecord;
+import htsjdk.samtools.cram.structure.SubstitutionMatrix;
+
+import java.util.List;
+
+/**
+ * CRAM compression decisions: given the data come up with a way to compress it.
+ */
+public interface CramCompression {
+	/**
+	 * Analyze the records and build a {@link CompressionHeader} object
+	 * describing how to compress the records.
+	 * 
+	 * @param records
+	 *            the data to be compressed
+	 * @param substitutionMatrix
+	 *            a matrix of base substitution frequencies
+	 * @param sorted
+	 *            if true the records are assumed to be sorted by alignment
+	 *            position
+	 * @return an object describing how the data should be compressed
+	 */
+	public CompressionHeader buildCompressionHeader(final List<CramCompressionRecord> records,
+													final SubstitutionMatrix substitutionMatrix, final boolean sorted);
```

Comment:
The `public` modifier is not required needed for an interface method.

Also, as visible on this line that goes off the github screen, please replace all the tabs in this diff and others and replace them with spaces.